src/sortStudents.ts

```diff
@@ -1,16 +1,68 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): object[] {
+  const copyOfStudents: object[] = students.map(
+    (student: object): object => {
+      return { ...student };
+    },
+  );
```

Comment:
```suggestion
  const copyOfStudents= [...students];
```
in this situation it is enough to do so

Reply:
In the procces i thought I will change properties but eventually i realized that i don't  have to but forgot to change it, will fix.